### PR TITLE
Ignore empty fields, change method name

### PR DESCRIPTION
**Pull request by gupsho**
#### Summary
Fix check for optional fields in Manifest.isValid().

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13761

### CODE REVIEW

File: model/manifest_test.go [hunk 1]
```diff
@@ -31,6 +31,37 @@ func TestIsValid(t *testing.T) {
 		{"SettingSchema error", &Manifest{Id: "com.company.test", HomepageURL: "http://someurl.com", SupportURL: "http://someotherurl.com", Version: "5.10.0", MinServerVersion: "5.10.8", SettingsSchema: &PluginSettingsSchema{
 			Settings: []*PluginSetting{{Type: "Invalid"}},
 		}}, true},
+		{"Empty homepageURL, supportURL, minServerVersion, version", &Manifest{
+			Id:               "com.company.test",
```
Comment (golangcibot) [inline model/manifest_test.go]: File is not `gofmt`-ed with `-s` (from `gofmt`)

File: model/manifest_test.go [hunk 2]
```diff
@@ -31,6 +31,37 @@ func TestIsValid(t *testing.T) {
...
+			Name:             "thename",
+			Description:      "thedescription",
+			Server: &ManifestServer{
+				Executable: "theexecutable",
+			},
+			Webapp: &ManifestWebapp{
+				BundlePath: "thebundlepath",
+				BundleHash: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
```
Comment (hanzei) [inline model/manifest_test.go]: Sorry, I missed this in the initial review: `BundleHash` should not be set my plugins. Would you please just drop it from all tests?

File: model/manifest.go [hunk 3]
```diff
@@ -322,26 +322,30 @@ func (m *Manifest) IsValid() error {
...
+		}
 	}
 
-	_, err2 := semver.Parse(m.MinServerVersion)
-	if err2 != nil {
-		return errors.Wrap(err2, "failed to parse MinServerVersion")
+	if m.MinServerVersion != "" {
+		_, err2 := semver.Parse(m.MinServerVersion)
+		if err2 != nil {
+			return errors.Wrap(err2, "failed to parse MinServerVersion")
```
Comment (hanzei) [inline model/manifest.go]: I think the different variable name is not longer needed here and also below.

File: model/manifest_test.go [hunk 4]
```diff
@@ -31,6 +31,37 @@ func TestIsValid(t *testing.T) {
 		{"SettingSchema error", &Manifest{Id: "com.company.test", HomepageURL: "http://someurl.com", SupportURL: "http://someotherurl.com", Version: "5.10.0", MinServerVersion: "5.10.8", SettingsSchema: &PluginSettingsSchema{
 			Settings: []*PluginSetting{{Type: "Invalid"}},
 		}}, true},
+		{"Empty homepageURL, supportURL, minServerVersion, version", &Manifest{
```
Comment (hanzei) [inline model/manifest_test.go]: Would you please add a test with the minimum of a manifest? This manifest should only contain a valid id.

**Comment by hanzei** [comment]
@gupsho Heads up that there is a merge conflict with your PR

**Comment by gupsho** [comment]
> @gupsho Heads up that there is a merge conflict with your PR

Fixed.

**Comment by hanzei** [comment]
@agnivade Could you please review this PR? @streamer45 is OOO.

**Comment by hanzei** [comment]
Skipping QA review as this change is not related to the product.
